magicblock-ledger/src/ledger_truncator.rs

```diff
@@ -281,49 +280,41 @@ impl<T: FinalityProvider> LedgerTrunctationWorker<T> {
         // Compaction can be run concurrently for different cf
         // but it utilizes rocksdb threads, in order not to drain
         // our tokio rt threads, we split the effort in just 3 tasks
-        let mut measure = Measure::start("Manual compaction");
-        let mut join_set = JoinSet::new();
-        join_set.spawn({
-            let ledger = ledger.clone();
-            async move {
-                ledger.compact_slot_range_cf::<Blocktime>(
-                    Some(from_slot),
-                    Some(to_slot + 1),
-                );
-                ledger.compact_slot_range_cf::<Blockhash>(
-                    Some(from_slot),
-                    Some(to_slot + 1),
-                );
-                ledger.compact_slot_range_cf::<PerfSamples>(
-                    Some(from_slot),
-                    Some(to_slot + 1),
-                );
-                ledger.compact_slot_range_cf::<SlotSignatures>(
-                    Some((from_slot, u32::MIN)),
-                    Some((to_slot + 1, u32::MAX)),
-                );
-            }
+        let ledger_copy = ledger.clone();
+        let handler1 = spawn_blocking(move || {
+            ledger_copy.compact_slot_range_cf::<Blocktime>(
+                Some(from_slot),
+                Some(to_slot + 1),
+            );
+            ledger_copy.compact_slot_range_cf::<Blockhash>(
+                Some(from_slot),
+                Some(to_slot + 1),
+            );
+            ledger_copy.compact_slot_range_cf::<PerfSamples>(
+                Some(from_slot),
+                Some(to_slot + 1),
+            );
+            ledger_copy.compact_slot_range_cf::<SlotSignatures>(
+                Some((from_slot, u32::MIN)),
+                Some((to_slot + 1, u32::MAX)),
+            );
         });
-
         // Can not compact with specific range
-        join_set.spawn({
-            let ledger = ledger.clone();
-            async move {
-                ledger.compact_slot_range_cf::<TransactionStatus>(None, None);
-                ledger.compact_slot_range_cf::<Transaction>(None, None);
-            }
+        let ledger_copy = ledger.clone();
+        let handler2 = spawn_blocking(move || {
```

Comment:
can't we just spawn the entire compaction on a single thread, instead of breaking it up between multiple, since there're limited number of cores, those extra threads (which are very hot) might be competing way too aggressively with other potentially more important threads